proposals/public-suffix.md

```diff
@@ -0,0 +1,824 @@
+extension to allow users to create containers that isolate organizational base domains:
+[PR #2352](https://github.com/mozilla/multi-account-containers/pull/2352).
+Currently, users must manually curate their containers to capture all anticipated subdomains
+of an organization's base domain, and this manual approach is cumbersome and error-prone.
+
+In addition to this, an analysis has been done of the relevant use cases of a selection of
+other extensions that currently roll their own solutions in order to parse the PSL, and the
+use cases are set out in the [Use Cases](#use-cases) section. Although the extension
+authors have not been consulted directly, it is assumed that they would stand to benefit
+from this API due to the reasons stated in the [Objective](#objective) section.
+
+| Extension | Relevant use cases | Description |
+|-----------|--------------------|-------------|
+| [multi-account-containers](https://addons.mozilla.org/en-US/firefox/addon/multi-account-containers/) | #1, #2 | isolate requests into containers by organizational base domain, view all domains grouped by organization |
+| [uBlock Origin](https://addons.mozilla.org/en-US/firefox/addon/ublock-origin/) | #1, #2, #3 | create organization/3rd-party request-blocking rules, view request blocking status by organization, create blocking rules for specifically 3rd-party requests |
+| [adblock-plus](https://addons.mozilla.org/en-US/firefox/addon/adblock-plus/) | #1, #3 | create organization/3rd-party request-blocking rules |
+| [duckduckgo-for-firefox](https://addons.mozilla.org/en-US/firefox/addon/duckduckgo-for-firefox/) | #3 | exclude requests from tracking detection unless they are 3rd-party |
+| [violentmonkey](https://addons.mozilla.org/en-US/firefox/addon/violentmonkey/) | #1 | run scripts automatically for requests having a specific organization name plus any eTLD |
+
+## Specification
+
+### Schema
+
+A new API `publicSuffix` is added as follows:
+
+```ts
+namespace publicSuffix {
+
+
+  // METHODS
+
+  // Determines if the given hostname is itself a known eTLD (i.e. in the PSL).
+  export function isKnownSuffix(
+    hostname: string,
+  )
+  : boolean;
+
+  // Gets the known eTLD, if any, of a given hostname.
+  export function getKnownSuffix(
+    hostname: string,
+  )
+  : string | null;
+
+  // Gets the registrable domain of a given hostname.
+  export function getDomain(
+    hostname: string,
+    options?: DomainOptions,
+  )
+  : string | null;
+
+  // Gets the value of the VERSION metadata field in the PSL dataset if available
+  export function getVersion(): string | null;
+
+
+  // INTERFACES
+
+  // Options that may be passed to the API method to control its behaviour.
+  interface DomainOptions {
+    // If true, the returned domain should be encoded as Unicode.
+    // Default = false (Punycode)
+    unicode?: boolean,
+    // If true, the returned domain may be an IP address.
```

Comment:
```suggestion
    // If true, the input hostname may be an IP address, and be returned as is.
```

Reply:
Addressed in e0f283531330fb48febfad686795793a41dc630f